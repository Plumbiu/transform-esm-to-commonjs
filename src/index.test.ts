import { expect, test } from 'vitest'
import transformModulesToCommonjs from '.'

const code = `
import React, { useEffect } from 'react'
import data from './bar.json' with { type: 'json' }
import * as vue from 'vue'

export default async function App() {
  const yaml = await import('yaml')
  return <div>app</div>
}

const foo: string = 'bar'
export { foo }

export const c = ''
`.trim()

test('transformModulesToCommonjs', () => {
  const result = transformModulesToCommonjs(code)
  expect(result).toMatchSnapshot()
})
