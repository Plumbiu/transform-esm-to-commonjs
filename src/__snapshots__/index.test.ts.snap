// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`transformModulesToCommonjs 1`] = `
"let __require__=(...args)=>new Promise((r)=>r(require(...args)));
let React=require('react');let {useEffect}=require('react');
let data=require('./bar.json');
let vue=require('vue');

export .default= async function App() {
  const yaml = await __require__('yaml')
  return <div>app</div>
}

const foo: string = 'bar'
exports.foo=foo;

exports.c=c;"
`;
